src/backends/kafka/mod.rs

```diff
@@ -205,17 +256,13 @@ impl<'a> ArroyoConsumer<'a, KafkaPayload> for KafkaConsumer {
     }
 
     fn tell(&self) -> Result<HashMap<Partition, u64>, ConsumerError> {
-        if [
-            KafkaConsumerState::Closed,
-            KafkaConsumerState::Error,
-            KafkaConsumerState::NotSubscribed,
-        ]
-        .contains(&self.state)
-        {
-            return Err(ConsumerError::ConsumerClosed);
+        match self.state {
+            KafkaConsumerState::Closed => Err(ConsumerError::ConsumerClosed),
+            KafkaConsumerState::NotSubscribed | KafkaConsumerState::Error => {
+                Err(ConsumerError::InvalidConsumerState)
+            }
```

Comment:
I think ConsumerClosed is a special case of InvalidConsumerState. Why not having `InvalidConsumerState` and passing the state as a parameter for the error when we raise it ?

Reply:
Done. I do worry it is excessively verbose to write `Err(ConsumerError::InvalidConsumerState(InvalidState::Closed))` and I do think it might be better to just add all 3 kinds of errors (not subscribed, error and closed) to the list instead. What do you think/

Reply:
Switched to the 3 errors option. I think it's more straightforward this way.